app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter
+    author.update(post_counter: author.posts.count)
+  end
+
```

Comment:
- [ ] It would be nicer to be calling the update_posts_counter every time a post is created and saved to the database. Make sure the after_save method is called just on top after the validates. And make the `update_post_counter` method private.
``` # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```